lib/media-controls-impl.js

```diff
@@ -0,0 +1,524 @@
+  function getTouchPos() {
+    return (!navigator.mozL10n ||
+             navigator.mozL10n.language.direction === 'ltr') ?
+       (clientX - self.sliderRect.left) :
+       (self.sliderRect.right - clientX);
+  }
+
+  var touchPos = getTouchPos();
+  var pos = touchPos / this.sliderRect.width;
+
+  pos = Math.max(pos, 0);
+  pos = Math.min(pos, 1);
+
+  // Update the slider to match the position of the user's finger.
+  // Note, however, that we don't update the displayed time until
+  // we actually get a 'seeked' event.
+  var percent = pos * 100 + '%';
+  this.els.playHead.classList.add('active');
+  this.movePlayHead(percent);
+  this.els.elapsedTime.style.width = percent;
+
+  this.mediaPlayer.fastSeek(this.mediaPlayer.duration * pos);
+};
+
+MediaControlsImpl.prototype.handleSliderMoveEnd = function() {
+
+  // If the user is not dragging the slider, noop
+  if (!this.dragging) {
+    return false;
+  }
+
+  this.dragging = false;
+  this.sliderRect = null;
+
+  this.els.playHead.classList.remove('active');
+
+  // If the media was playing when the user began dragging the slider
+  // (and the slider was not dragged to the end), begin playing the
+  // media.
+  if (!this.isPausedWhileDragging &&
+      this.mediaPlayer.currentTime !== this.mediaPlayer.duration) {
+    this.mediaPlayer.play();
+  }
+};
+
+MediaControlsImpl.prototype.formatTime = function(duration) {
+  function padLeft(num, length) {
+    var r = String(num);
+    while (r.length < length) {
+      r = '0' + r;
+    }
+    return r;
+  }
+
+  duration = Math.round(duration);
+  var minutes = Math.floor(duration / 60);
+  var seconds = duration % 60;
+  if (minutes < 60) {
+    return padLeft(minutes, 2) + ':' + padLeft(seconds, 2);
+  }
+  var hours = Math.floor(minutes / 60);
+  minutes = Math.floor(minutes % 60);
+  return hours + ':' + padLeft(minutes, 2) + ':' + padLeft(seconds, 2);
+};
+
+MediaControlsImpl.prototype.seekBy = function(seekTime) {
+  //
+  // If seeking will move the media position before the beginning or past
+  // the end, stop the auto-seeking (if in progress) and position the media
+  // accordingly.
+  //
+  if (seekTime >= this.mediaPlayer.duration || seekTime < 0) {
+    if (this.intervalId) {
+       window.clearInterval(this.intervalId);
+       this.intervalId = null;
+    }
+
+    if (seekTime >= this.mediaPlayer.duration) {
+      seekTime = this.mediaPlayer.duration;
+      // If the user tries to seek past the end of the media while the media
+      // is playing, pause the playback.
+      //
+      // Also, set a flag so the media controls will know the media wasn't
+      // played until the end and therefore does not skip back to the
+      // beginning.
+      if (!this.mediaPlayer.paused) {
+        this.mediaPlayer.pause();
+        this.pausedAtEndWhilePlaying = true;
+      }
+    }
+    else {
+      seekTime = 0;
+    }
+  }
+
+  this.mediaPlayer.fastSeek(seekTime);
+};
+
+MediaControlsImpl.prototype.unload = function(e) {
```

Comment:
I'd just call this detach() to match the name in the parent MediaControls class.